stage0/src/sev.rs

```diff
@@ -28,14 +28,37 @@ use oak_sev_guest::{
 use spinning_top::Spinlock;
 use x86_64::{
     instructions::tlb,
-    registers::control::Cr3,
+    registers::{control::Cr3, model_specific::Msr},
     structures::paging::{
         frame::PhysFrameRange, page::AddressNotAligned, Page, PageSize, PageTable, PageTableFlags,
         PhysFrame, Size1GiB, Size2MiB, Size4KiB,
     },
     PhysAddr, VirtAddr,
 };
 
+/// The identifier for the MSR used to set the default caching mode.
+const IA32_MTRR_DEFTYPE: u32 = 0x2ff;
+
+/// Value for enanabling MTRR with write-protect mode by default.
+const MTRR_EN_WITH_WP: u64 = 0x805;
+
+/// Write a config value to IA32_MTRR_DefTpype MSR register to set the default caching mode to
+/// write-protected (WP).  This is required by Linux kernels since July, 2022, with this
+/// requirement back-ported to 5.15.X.
+///
+/// The Linux kernel gives a warning that MTRR is not setup properly, and that BOIOS proably
+/// only set the MTRR mode on one CPU, which is the case, since we do not want to deal with
+/// the complexity of setting up multiple cores.  The Linux kernel propagates the MTRR set
+/// on this core to the others, and reports that MTRR settings have been fixed.
+pub fn enable_mtrr() {
```

Comment:
It's a bit overkill, but let's do things properly :)

Follow the pattern `x86_64` crate does for these MSRs: https://docs.rs/x86_64/latest/x86_64/registers/model_specific/index.html

That is, have a `struct Mtrr` that embeds the raw `Msr` data structure, and exposes some higher-level methods, such as `enable(default_type: MemoryType)` , with MemoryType being an enum that lists (at the very least) the WriteProtect value (5).

Reply:
Will do.  It will also need to set bit 11, which enables MTRR.

Reply:
Done, though I lack confidence in Rust.